ONSdigital/eq-author-app
# ✨ EAR 1407 - `Firestore`: split sections out into subcollection

### What is the context of this PR?

[Ticket](https://collaborate2.ons.gov.uk/jira/browse/EAR-1407)

⚠️ **Firestore only**

To avoid hitting the Firestore document size limit it was decided to move `sections` out into their own subcollection. This PR migrates questionnaires to this new format on save.

The new structure is as follows:

```
questionnaires
  |-- <questionnaire ID>
      |-- versions
          |-- <version ID>
               |-- sections
                    |-- <section ID>
                    |-- <section ID>
```

### How to review

- Edit your `docker-compose.yml` file for `eq-author-api` to specify `DATABASE: firestore`.
- Re-start author-api, create a questionnaire
- Try importing an existing questionnaire
- Create sections, should work as before
- Move sections, should work as before
- If feeling adventurous, debug and/or query state of the Firestore DB using node API to verify the new structure

### What to do after everything is green

1. - [ ] Bring this branch up-to-date with Origin/Master
2. - [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. - [ ] Click the **Merge** button on this pull request
4. - [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. - [ ] Move the Jira ticket for this task into the next stage of the process
